src/test/java/io/cdap/delta/datastream/DatastreamTableRegistryTest.java

```diff
@@ -17,74 +16,66 @@
+    Mockito.when(schema.getSchema()).thenReturn("schema");
+    Mockito.when(schema.getOracleTablesList()).thenReturn(Arrays.asList(table));
+
+    OracleRdbms oracleRdbms = Mockito.mock(OracleRdbms.class);
+    Mockito.when(response.getOracleRdbms()).thenReturn(oracleRdbms);
+    Mockito.when(oracleRdbms.getOracleSchemasList()).thenReturn(Arrays.asList(schema));
+
+    DatastreamClient datastreamClient = Mockito.mock(DatastreamClient.class);
+    Mockito.when(datastreamClient.discoverConnectionProfile(Mockito.any())).thenReturn(response);
+
+    DatastreamTableRegistry datastreamTableRegistry = new DatastreamTableRegistry(datastreamConfig, datastreamClient);
+    TableList tableList = datastreamTableRegistry.listTables();
 
-  private void testListDesribeTable(boolean usingExisting) throws Exception {
-    TableList tableList = null;
-    DatastreamDeltaSource deltaSource = createDeltaSource(usingExisting);
-    try (TableRegistry registry = deltaSource.createTableRegistry(null)) {
-      tableList = registry.listTables();
-    }
     assertNotNull(tableList);
     List<TableSummary> tables = tableList.getTables();
     assertNotNull(tables);
     assertFalse(tables.isEmpty());
     System.out.println("table counts:" + tables.size());
-    for (TableSummary table : tables) {
-      assertNotNull(table);
-      assertEquals(oracleDb, table.getDatabase());
-      String schema = table.getSchema();
-      assertNotNull(schema);
-      String tableName = table.getTable();
-      assertNotNull(tableName);
-      System.out.println(String.format("table : %s.%s", schema, tableName));
-      TableDetail tableDetail = null;
-      try (TableRegistry registry = deltaSource.createTableRegistry(null)) {
-        tableDetail = registry.describeTable(oracleDb, schema, tableName);
-      }
-      assertNotNull(tableDetail);
-      List<ColumnDetail> columns = tableDetail.getColumns();
-      assertNotNull(columns);
-      assertFalse(columns.isEmpty());
-      for (ColumnDetail column : columns) {
-        assertNotNull(column);
-        assertNotNull(column.getName());
-        Map<String, String> properties = column.getProperties();
-        assertNotNull(properties);
-        assertNotNull(column.getType());
-        System.out.println(String.format("column: %s type : %s", column.getName(), column.getType()));
-      }
-      List<String> primaryKeys = tableDetail.getPrimaryKey();
-      System.out.println("primary keys: " + primaryKeys);
-      assertNotNull(primaryKeys);
-      for (String key : primaryKeys) {
-        assertNotNull(key);
-        assertFalse(key.isEmpty());
-      }
-    }
+
+    //verify hierarchy depth set to 2
+    ArgumentCaptor<DiscoverConnectionProfileRequest> captor = ArgumentCaptor.forClass(
+      DiscoverConnectionProfileRequest.class);
+    Mockito.verify(datastreamClient).discoverConnectionProfile(captor.capture());
+    assertEquals(captor.getValue().getHierarchyDepth(), 2);
```

Comment:
Expected value should be the first argument
```suggestion
    assertEquals(2, captor.getValue().getHierarchyDepth());
```